hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java

```diff
@@ -171,13 +171,15 @@ public static void main(String[] args) {
       System.exit(1);
     }
     final JavaSparkContext jsc = UtilHelpers.buildSparkContext("compactor-" + cfg.tableName, cfg.sparkMaster, cfg.sparkMemory);
+    int ret = 0;
     try {
       HoodieCompactor compactor = new HoodieCompactor(jsc, cfg);
-      compactor.compact(cfg.retry);
+      ret = compactor.compact(cfg.retry);
     } catch (Throwable throwable) {
       LOG.error("Fail to run compaction for " + cfg.tableName, throwable);
     } finally {
       jsc.stop();
+      System.exit(ret);
```

Comment:
why we need call System.exit here ?

Reply:
Thank you for your review. Here you can find out why. https://issues.apache.org/jira/browse/HUDI-3945